src/signum.rs

```diff
@@ -12,6 +12,31 @@ pub enum Sgn0Result {
     Negative,
 }
 
+impl From<bool> for Sgn0Result {
+    fn from(val: bool) -> Self {
+        if val {
+            // Negative values = 1
+            Sgn0Result::Negative
+        } else {
+            // Non negative values = 0
+            Sgn0Result::NonNegative
+        }
+    }
+}
+
+impl BitAnd for Sgn0Result {
+    type Output = Self;
+
+    fn bitand(self, rhs: Self) -> Self::Output {
+        if self == rhs && self == Sgn0Result::Negative {
+            // 1 & 1 == 1
+            Sgn0Result::Negative
```

Comment:
I trust it is working but the logic here evades me: this is saying that a Negative AND NonNegative = NonNegative, right ?
Isn't that weird ? Or do we only say a Fq2 is negative when both Fq are negative ? Might be, i'm just asking questions here from curiosity.